src/com/owncloud/android/files/services/FileUploader.java

```diff
@@ -209,6 +211,21 @@ public void onDestroy() {
     public int onStartCommand(Intent intent, int flags, int startId) {
         Log_OC.d(TAG, "Starting command with id " + startId);
 
+        if (intent.hasExtra(KEY_CANCEL_ALL) && intent.hasExtra(KEY_ACCOUNT)){
+            Account account = intent.getParcelableExtra(KEY_ACCOUNT);
+
+            Log_OC.d(TAG, "Account= " + account.name);
+
+            if (mCurrentUpload != null) {
```

Comment:
This block is duplicating the code in FileUploaderBinder.cancel(Account account); should be safe call mBinder.cancel(account) instead. 

Besides, should explicltly return after cacelUploadForAccount(account.name) with Service.START_NOT_STICKY. Right now the exit is done in the next if block, as a side effect of the extras not included in the Intent.  (I know this method is a mess with so many return points, all of them with the same value, but let's not rewrite that here, since there are many changes in FileUploader in the branch about the list of uploads).